sklearn/metrics/_pairwise_distances_reduction/_base.pxd.tp

```diff
@@ -17,19 +17,30 @@ cimport numpy as cnp
 
 from cython cimport final
 
-from ...utils._typedefs cimport ITYPE_t, DTYPE_t
+from ...utils._typedefs cimport ITYPE_t, DTYPE_t, SPARSE_INDEX_TYPE_t
 
 cnp.import_array()
 
 {{for name_suffix, INPUT_DTYPE_t, INPUT_DTYPE in implementation_specific_values}}
 
 from ._datasets_pair cimport DatasetsPair{{name_suffix}}
 
-cpdef DTYPE_t[::1] _sqeuclidean_row_norms{{name_suffix}}(
+cpdef DTYPE_t[::1] _sqeuclidean_row_norms{{name_suffix}}_dense(
     const {{INPUT_DTYPE_t}}[:, ::1] X,
     ITYPE_t num_threads,
 )
 
+cpdef DTYPE_t[::1] _sqeuclidean_row_norms{{name_suffix}}_sparse(
+    const {{INPUT_DTYPE_t}}[:] X_data,
+    const SPARSE_INDEX_TYPE_t[:] X_indptr,
+    ITYPE_t num_threads,
+)
```

Comment:
We probably do not need to expose those two declarations now that `_sqeuclidean_row_norms{{name_suffix}}` is the interface dispatching to either.